parsers.py

```diff
@@ -99,22 +93,95 @@ def main(self, google_maps_url: str) -> list[RenovationLead]:
 
         for block in self.get_companies_blocks():
             renovation_leads.append(
-                self.create_renovation_leads_instances(block)
+                self.create_renovation_lead_instance(block)
             )
 
         return renovation_leads
 
 
 class OwnersParser(ChromeParser):
-    pass
+    @staticmethod
+    def search_url(company_name: str) -> str:
+        return f"https://find-and-update.company-information.service.gov.uk/search?q={company_name.replace(' ', '+')}"
+
+    @staticmethod
+    def is_half_similar(string1: str, string2) -> bool:
```

Comment:
TODO: use more accurate names. For instance, current_company_name & first_link_company_name